src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/JavaScriptEncoder.cs

```diff
@@ -84,35 +76,31 @@ public DefaultJavaScriptEncoder(params UnicodeRange[] allowedRanges) : this(new
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public override bool WillEncode(int unicodeScalar)
         {
-            if (UnicodeHelpers.IsSupplementaryCodePoint(unicodeScalar)) return true;
+            if (UnicodeHelpers.IsSupplementaryCodePoint(unicodeScalar))
+                return true;
             return !_allowedCharacters.IsUnicodeScalarAllowed(unicodeScalar);
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public unsafe override int FindFirstCharacterToEncode(char* text, int textLength)
         {
             if (text == null)
-            {
                 throw new ArgumentNullException(nameof(text));
-            }
-            return _allowedCharacters.FindFirstCharacterToEncode(text, textLength);
+            return FindFirstCharacterToEncode(new ReadOnlySpan<char>(text, textLength));
         }
 
         // The worst case encoding is 6 output chars per input char: [input] U+FFFF -> [output] "\uFFFF"
         // We don't need to worry about astral code points since they're represented as encoded
         // surrogate pairs in the output.
-        public override int MaxOutputCharactersPerInputCharacter
-        {
-            get { return 12; } // "\uFFFF\uFFFF" is the longest encoded form 
-        }
+        public override int MaxOutputCharactersPerInputCharacter => 12; // "\uFFFF\uFFFF" is the longest encoded form 
 
-        static readonly char[] s_b = new char[] { '\\', 'b' };
-        static readonly char[] s_t = new char[] { '\\', 't' };
-        static readonly char[] s_n = new char[] { '\\', 'n' };
-        static readonly char[] s_f = new char[] { '\\', 'f' };
-        static readonly char[] s_r = new char[] { '\\', 'r' };
-        static readonly char[] s_forward = new char[] { '\\', '/' };
-        static readonly char[] s_back = new char[] { '\\', '\\' };
+        private static readonly char[] s_b = new char[] { '\\', 'b' };
```

Comment:
I've seen a few PRs recently to change private static arrays to private static readonly ReadOnlySpan,  primarily for byte though. Does this work here too? e.g. PR #20768

Reply:
This Span optimization works for `byte` only.

It can be used here. These pre-initialized blobs can be changed to `ReadOnlySpan<byte>` and then copies over one byte at a time into the destination. Since they are always very small, having a local copy that copies one byte at a time would be faster than calling TryCopyTo.